pkg/storage/storage_test.go

```diff
@@ -278,87 +276,117 @@ func TestStorageHistory(t *testing.T) {
 	}
 }
 
-func TestStorageRemoveLeastRecentWithError(t *testing.T) {
-	storage := Init(driver.NewMemory())
-	storage.Log = t.Logf
-
-	storage.MaxHistory = 1
-
-	const name = "angry-bird"
-
-	// setup storage with test releases
-	setup := func() {
-		// release records
-		rls1 := ReleaseTestData{Name: name, Version: 1, Status: rspb.StatusSuperseded}.ToRelease()
-
-		// create the release records in the storage
-		assertErrNil(t.Fatal, storage.Driver.Create(makeKey(rls1.Name, rls1.Version), rls1), "Storing release 'angry-bird' (v1)")
-	}
-	setup()
-
-	rls2 := ReleaseTestData{Name: name, Version: 2, Status: rspb.StatusSuperseded}.ToRelease()
-	wantErr := driver.ErrNoDeployedReleases
-	gotErr := storage.Create(rls2)
-	if !errors.Is(gotErr, wantErr) {
-		t.Fatalf("Storing release 'angry-bird' (v2) should return the error %#v, but returned %#v", wantErr, gotErr)
-	}
-}
+//func TestStorageRemoveLeastRecentWithError(t *testing.T) {
```

Comment:
I commented this test as the PR changes the error conditions, so this error won't happen anymore.
if the reviewers are happy with this PR I will delete the commented test.